tests/refs/iterator.c

```diff
@@ -15,6 +15,7 @@ void test_refs_iterator__cleanup(void)
 }
 
 static const char *refnames[] = {
+	"refs/blobs/annotated_tag_to_blob",
```

Comment:
This thing here was missing to complete my picture of your issue. I didn't realize you had tags in non-standard locations. Anyway, we should definitly be able to cope with that as users do strange things.